lib/src/models/parsed_network.dart

```diff
@@ -1,13 +1,23 @@
 import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:ledger_cardano/src/utils/cardano_networks.dart';
 
 part 'parsed_network.freezed.dart';
 
 @freezed
 sealed class ParsedNetwork with _$ParsedNetwork {
   ParsedNetwork._();
 
-  factory ParsedNetwork({
-    required int protocolMagic,
-    required int networkId,
-  }) = _ParsedNetwork;
+  factory ParsedNetwork.testnet() = ParsedNetworkTestnet;
+
+  factory ParsedNetwork.mainnet() = ParsedNetworkMainnet;
+    
+  late final int protocolMagic = switch(this) {
+    ParsedNetworkMainnet() => CardanoNetwork.mainnet.protocolMagic,
+    ParsedNetworkTestnet() => CardanoNetwork.testnet.protocolMagic,
+  };
+  late final int networkId = switch(this) {
+    ParsedNetworkMainnet() => CardanoNetwork.mainnet.networkId,
+    ParsedNetworkTestnet() => CardanoNetwork.testnet.networkId,
+  };
+  
 }
```

Comment:
This sealed class is actually not needed. Same applies for the one in `network.dart`. The `CardanoNetwork` can be used directly where needed (using freezed here instead of enum brings no value)

Reply:
done